### PR TITLE
Reports sprint

### PR DESCRIPTION
**Pull request by Talgaon4**
Reports sprint

### CODE REVIEW

File: EKrut-Server/src/ekrut/server/db/ReportDAO.java [hunk 1]
```diff
@@ -215,60 +223,73 @@ public Report fetchCustomerReportByID(int reportID) {
...
-			}
-			else {
-				return null; 
+			
+			//fetch top sellers
+			Map<String, Integer> topSellersData = new HashMap<>();
+			
+			while(rs3.next()) {
+				topSellersData.put(rs2.getString("itemName"),
+						rs2.getInt("totalOrders"));
```
Comment (almogkh) [inline EKrut-Server/src/ekrut/server/db/ReportDAO.java]: Should these be `rs3`?

Comment (Talgaon4) [reply EKrut-Server/src/ekrut/server/db/ReportDAO.java]: yes

File: EKrut-Server/src/ekrut/server/db/ReportDAO.java [hunk 2]
```diff
@@ -524,26 +548,39 @@ public boolean createOrderReport(Report report) {
 		Integer reportID = report.getReportID();
 		
 		try {
+			//Insert topSellers
+			for (Map.Entry<String, Integer> entry : report.getTopSellersData().entrySet()) {
+				ps1.setInt(1,  reportID);
+				ps1.setString(2, entry.getKey());
+				ps1.setInt(2, entry.getValue());
```
Comment (almogkh) [inline EKrut-Server/src/ekrut/server/db/ReportDAO.java]: This should be 3.

File: EKrut-Server/src/ekrut/server/db/ReportDAO.java [hunk 3]
```diff
@@ -524,26 +548,39 @@ public boolean createOrderReport(Report report) {
...
-				ps2.setInt(3, entry.getValue());
-				ps2.addBatch();
+				ps2.setInt(3, entry.getValue().get(0));
+				ps2.setInt(4, entry.getValue().get(1));
+				ps2.setInt(5, entry.getValue().get(2));
+				ps2.setInt(6, entry.getValue().get(3));
+				ps2.setInt(7, entry.getValue().get(4));
+				ps2.setInt(8, entry.getValue().get(5));
+				ps2.setInt(9, entry.getValue().get(6));
+				ps2.setInt(9, entry.getValue().get(7));
```
Comment (almogkh) [inline EKrut-Server/src/ekrut/server/db/ReportDAO.java]: This should be 10?

Comment (Talgaon4) [reply EKrut-Server/src/ekrut/server/db/ReportDAO.java]: yes, fixed

File: EKrut-Server/src/ekrut/server/db/ReportDAO.java (hunk 2 above)
Comment (Talgaon4) [reply EKrut-Server/src/ekrut/server/db/ReportDAO.java]: fixed

**Pull request merged**